### PR TITLE
Fix `to_graph` DataPipeGraph visualization function

### LINKED ISSUE #869: datapipes.to_graph() throws AttributeError: 'tuple' object has no attribute 'items'
### 🐛 Describe the bug

I upgraded to `torch v1.13.0` & `torchdata v0.5.0` over the weekend and am getting `AttributeError` for  `datapipes.to_graph()` call after that - it was working completely fine before the upgrade.

Sample code to replicate the issue
```python
from torchdata.datapipes.iter import IterableWrapper
from torchdata.datapipes.utils import to_graph

items = IterableWrapper(range(10))
items.map(lambda x: x * 2)

print(list(items))
# [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] prints the right result

to_graph(items)
# AttributeError: 'tuple' object has no attribute 'items'
```

Complete stack trace:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [24], line 9
      5 items.map(lambda x: x * 2)
      7 print(list(items))
----> 9 to_graph(items)

File ~/.local/share/hatch/env/virtual/ml-pipeline-HWwNZ1N1/nb/lib/python3.9/site-packages/torchdata/datapipes/utils/_visualization.py:175, in to_graph(dp, debug)
    164 node_attr = dict(
    165     style="filled",
    166     shape="box",
   (...)
    171     fontname="monospace",
    172 )
    173 graph = graphviz.Digraph(node_attr=node_attr, graph_attr=dict(size="12,12"))
--> 175 for node in to_nodes(dp, debug=debug):
    176     fillcolor: Optional[str]
    177     if not node.parents:

File ~/.local/share/hatch/env/virtual/ml-pipeline-HWwNZ1N1/nb/lib/python3.9/site-packages/torchdata/datapipes/utils/_visualization.py:120, in to_nodes(dp, debug)
    116                 actual_child.add_parent(fixed_parent_node)
    118     return nodes
--> 120 return aggregate(recurse(traverse_dps(dp)))

File ~/.local/share/hatch/env/virtual/ml-pipeline-HWwNZ1N1/nb/lib/python3.9/site-packages/torchdata/datapipes/utils/_visualization.py:69, in to_nodes.<locals>.aggregate(nodes)
     67 def aggregate(nodes):
     68     groups = defaultdict(list)
---> 69     for node in nodes:
     70         groups[node].append(node)
     72     nodes = set()

File ~/.local/share/hatch/env/virtual/ml-pipeline-HWwNZ1N1/nb/lib/python3.9/site-packages/torchdata/datapipes/utils/_visualization.py:65, in to_nodes.<locals>.recurse(dp_graph, child)
     63     node.add_child(child)
     64 yield node
---> 65 yield from recurse(dp_parents, child=node)

File ~/.local/share/hatch/env/virtual/ml-pipeline-HWwNZ1N1/nb/lib/python3.9/site-packages/torchdata/datapipes/utils/_visualization.py:60, in to_nodes.<locals>.recurse(dp_graph, child)
     59 def recurse(dp_graph, child=None):
---> 60     for dp_node, dp_parents in dp_graph.items():
     61         node = Node(dp_node)
     62         if child is not None:

AttributeError: 'tuple' object has no attribute 'items'
```

### Versions

Collecting environment information...
PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.22.4
Libc version: glibc-2.27

Python version: 3.9.12 (main, Apr  5 2022, 06:56:58)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-1075-aws-x86_64-with-glibc2.27
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.4
[pip3] pytorch-lightning==1.7.7
[pip3] torch==1.13.0
[pip3] torchdata==0.5.0
[pip3] torchmetrics==0.10.1
[pip3] torchvision==0.14.0
[conda] blas                      1.0                         mkl  
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py39h7f8727e_0  
[conda] mkl_fft                   1.3.1            py39hd3c417c_0  
[conda] mkl_random                1.2.2            py39h51133e4_0  
[conda] numpy                     1.21.5           py39he7a7128_1  
[conda] numpy-base                1.21.5           py39hf524024_1  
[conda] numpydoc                  1.2                pyhd3eb1b0_0

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #872

Fixes #869 and adds tests to ensure `to_graph` can execute without raising exception

Differential Revision: [D40956306](https://our.internmc.facebook.com/intern/diff/D40956306)

### CODE REVIEW

File: test/test_graph.py [hunk 1]
```diff
@@ -204,5 +212,33 @@ def test_multiprocessing_reading_service(self) -> None:
         self.assertEqual(d1, d2)
 
 
+class TestGraphVisualization(expecttest.TestCase):
+    @unittest.skipIf(not HAS_GRAPHVIZ, "Package `graphviz` is required to test graph visualization functionalities.")
```
Comment (NivekT) [inline test/test_graph.py:216]: @ejguan Do I need to make changes to CI to make sure the package is installed?

File: test/test_graph.py [hunk 2]
```diff
@@ -204,5 +212,33 @@ def test_multiprocessing_reading_service(self) -> None:
...
+            {"dp": cdp1, "debug": True},
+            {"dp": dp4},
+            {"dp": dp4, "debug": True},
+            {"dp": cdp3, "debug": True},
+            {"dp": dp5},
+            {"dp": dp5, "debug": True},
+        ]
+        for kwargs in kwargs_list:
+            g = to_graph(**kwargs)
+            self.assertTrue(isinstance(g, graphviz.Digraph))
```
Comment (NivekT) [inline test/test_graph.py:240]: It is hard to verify the correctness of the graph object. Let me know if there is any better ideas.

Comment (ejguan) [reply test/test_graph.py:240]: Let's just validate there would be no Error if it's too hard.

File: test/test_graph.py (hunk 1 above)
Comment (ejguan) [reply test/test_graph.py:216]: No need. You can just add this package to https://github.com/pytorch/data/blob/main/test/requirements.txt

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40956306).

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40956306).

**Pull request closed without merging**